src/Symfony/Component/DependencyInjection/Container.php

```diff
@@ -426,6 +431,34 @@ protected function getEnv($name)
         return $this->envCache[$name] = $this->getParameter("env($name)");
     }
 
+    /**
+     * Returns the case sensitive id used at registration time.
+     *
+     * @param string $id
+     *
+     * @return string
+     *
+     * @internal
+     */
+    public function normalizeId($id)
+    {
+        if (!is_string($id)) {
+            $type = is_object($id) ? get_class($id) : gettype($id);
+            $id = (string) $id;
+            @trigger_error(sprintf('Non-string service identifiers are deprecated since Symfony 3.3 and won\'t be supported in 4.0 for service "%s" ("%s" given.) Cast it to string beforehand.', $id, $type), E_USER_DEPRECATED);
```

Comment:
At this point we dont really know what the case of  the non-string-service-id is right? Meaning this deprecation can be a false positive?

Ie. `$normalizedId  = strtolower((string) $id)` looks sufficient? Triggering deprecation as usual from below..